test/unit/unit_socket.c

```diff
@@ -39,13 +39,13 @@ START_TEST (test_socket)
     s = pico_socket_open(PICO_PROTO_IPV4, 99, NULL);
     fail_if(s != NULL, "Error got socket wrong parameters");
 
-    s = pico_socket_open(PICO_PROTO_IPV4, -109, NULL);
+    s = pico_socket_open(PICO_PROTO_IPV4, (uint16_t)-109, NULL);
```

Comment:
Not sure if simply deleting the test is a good idea. The negative number becomes an absurdly large number - are there other test cases checking for that? If there aren't, maybe just modify the test to use a large number, so that the test makes sense.